compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -894,6 +899,19 @@ public void visit(BLangImportPackage importPkgNode) {
 
         PackageID pkgId = new PackageID(orgName, nameComps, version);
 
+        // Un-exported modules not inside current package is not allowed to import.
+        BPackageSymbol bPackageSymbol = this.packageCache.getSymbol(pkgId);
+        if (bPackageSymbol != null && this.env.enclPkg.moduleContext != null) {
+            boolean isCurrentPackageModuleImport =
+                    this.env.enclPkg.moduleContext.descriptor().org() == bPackageSymbol.descriptor.org()
+                            && this.env.enclPkg.moduleContext.descriptor().packageName() == bPackageSymbol.descriptor
+                            .packageName();
+            if (!isCurrentPackageModuleImport && !bPackageSymbol.exported) {
+                dlog.error(importPkgNode.pos, DiagnosticErrorCode.MODULE_NOT_FOUND,
```

Comment:
Should be return from here as we do for annotation package ?

Reply:
Added a return after adding the diagnostic